cave/reader/smac2_reader.py

```diff
@@ -0,0 +1,195 @@
+                self.scen.train_insts[x-1])
+        data["seed"] = csv_data["Seed"]
+        data["time"] = csv_data["Runtime"]
+        if self.scen.run_obj == 'runtime':
+            data["cost"] = csv_data["Runtime"]
+        else:
+            data["cost"] = csv_data["Run Quality"]
+        data["status"] = csv_data["Run Result"]
+
+        # Load configurations
+        with open(configs_fn, 'r') as csv_file:
+            csv_data = list(csv.reader(csv_file, delimiter=',',
+                                       skipinitialspace=True))
+        configurations = {}  # id to config
+        for row in csv_data:
+            config_id = int(re.match(r'^(\d*):', row[0]).group(1))
+            params = [re.match(r'^\d*: (.*)', row[0]).group(1)]
+            params.extend(row[1:])
+            #self.logger.debug(params)
+            matches = [re.match(r'(.*)=\'(.*)\'', p) for p in params]
+            values = {m.group(1) : m.group(2) for m in matches}
+            rs = np.random.RandomState()
+            for name, value in values.items():
+                if isinstance(cs.get_hyperparameter(name), IntegerHyperparameter):
+                    values[name] = int(value)
+                if isinstance(cs.get_hyperparameter(name), FloatHyperparameter):
+                    values[name] = float(value)
+            configurations[config_id] = self._remove_inactive(cs, Configuration(cs,
+                                                    values=values,
+                                                    allow_inactive_with_values=True),
+                                                    values)
+        self.configurations = configurations
+
+        rh = CSV2RH().read_csv_to_rh(data,
+                                     pcs=cs,
+                                     configurations=configurations)
+
+        return (rh, validated_rh)
+
+    def get_trajectory(self, cs):
+        traj_fn = os.path.join(self.folder, '../traj-run-1.txt')
+        with open(traj_fn, 'r') as csv_file:
+            csv_data = list(csv.reader(csv_file, delimiter=',',
+                                       skipinitialspace=True))
+        header, csv_data = csv_data[0][:-1], np.array([csv_data[1:]])[0]
+        csv_data = pd.DataFrame(np.delete(csv_data, np.s_[5:], axis=1), columns=header)
+        csv_data = csv_data.apply(pd.to_numeric, errors='ignore')
+        traj = []
+        def add_to_traj(row):
+            new_entry = {}
+            new_entry['cpu_time'] = row['CPU Time Used']
+            new_entry['total_cpu_time'] = None
+            new_entry["wallclock_time"] = row['Wallclock Time']
+            new_entry["evaluations"] = -1
+            new_entry["cost"] = row["Estimated Training Performance"]
+            new_entry["incumbent"] = self.configurations[row["Incumbent ID"]]
+            traj.append(new_entry)
+        csv_data.apply(add_to_traj, axis=1)
+        return traj
+
+    def _remove_inactive(self, cs, configuration, values):
```

Comment:
Why is this necessary? Couldn't this be achived with the ConfigSpace package?

Reply:
Yes, I realized that later and forgot to change it.